src/service/extensions-impl.js

```diff
@@ -140,13 +142,29 @@ export class Extensions {
    * @restricted
    */
   registerExtension(extensionId, version, latest, factory, arg) {
-    const holder = this.getExtensionHolder_(extensionId, /* auto */ true);
+    const latestHolder = latest
+      ? this.extensions_[extensionKey(extensionId, LATEST_VERSION)]
+      : null;
+    const holder = this.getExtensionHolder_(
+      extensionId,
+      version,
+      latestHolder?.auto ?? true
```

Comment:
This is the only part that confuses me. I think we're trying to propagate an auto-insert for `latest` if we later try to register `0.1`? But why do that?

Reply:
This preserves the old behavior and IMO the right thing to do: if the `latest` version was added by the script and is not the "auto install" extension, then when it's loaded and registered here, it should also not be "auto install". Otherwise a FIE that includes `https://cdn.ampproject.org/v0/amp-ext-latest.js` would always cause the parent page to auto-install it because it always arrives as numerical (e.g. `0.2`).

To clarify, let's look at how a numerical sequence looks like:

1. A FIE is asking for `amp-ext:0.2`. It's immediately marked as non-auto-install extension.
2. Once downloaded, the extension registered with the call `registerExtension('amp-ext', '0.2', true|false)`. At this point the `Extensions` service knows that it's not an auto-install extension and it's NOT installed on the parent doc.

For a `latest` version the sequence looks like this:

1. A FIE is asking for `amp-ext:latest`. The `latest` version is immediately marked as non-auto-install extension. But we don't know what numerical version it belongs to.
2. Once downloaded, the extension registered with the call `registerExtension('amp-ext', '0.3', true)`. At this point the `Extensions` service doesn't know anything about `exp-ext:0.3` so it defaults to auto-install which would install it on the parent doc. This is not good and what's fixed here by taking the auto-install flag from the `latest` version when available. Notice that this is only done if the `0.3` has not been registered independently. Otherwise it's auto-install flag is used.

Reply:
Also, added an abbreviated version of this to the code comments. And there are several tests dedicated to this nuance.